src/main/java/com/google/devtools/build/lib/analysis/skylark/SkylarkActionFactory.java

```diff
@@ -544,15 +544,11 @@ private void registerStarlarkAction(
     if (EvalUtils.toBoolean(useDefaultShellEnv)) {
       builder.useDefaultShellEnvironment();
     }
-    if (executionRequirementsUnchecked != Runtime.NONE) {
-      builder.setExecutionInfo(
-          TargetUtils.filter(
-              SkylarkDict.castSkylarkDictOrNoneToDict(
-                  executionRequirementsUnchecked,
-                  String.class,
-                  String.class,
-                  "execution_requirements")));
-    }
+
+    Map<String, String> executionInfo =
+        TargetUtils.getFilteredExecutionInfo(executionRequirementsUnchecked, ruleContext.getRule());
```

Comment:
Does it make sense to do the validation here and pass "checked" execution requirements to the TargetUtils method?

Reply:
I don't have a strong opinion on this, but IMO it's better to keep it all in one place. And it's easier to test is this way ;)